src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -149,63 +54,11 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
   const adapter = chainAdapterManager.get(
     ethAsset.chainId,
   ) as ChainAdapter<KnownChainIds.EthereumMainnet>
-
-  const [lpLoading, setLpLoading] = useState<boolean>(true)
-  const [lpFoxBalance, setLpFoxBalance] = useState<string | null>(null)
-  const [lpEthBalance, setLpEthBalance] = useState<string | null>(null)
   const [ongoingTxId, setOngoingTxId] = useState<string | null>(null)
-  const [foxEthLpOpportunity, setFoxEthLpOpportunity] = useState<EarnOpportunityType>(lpOpportunity)
+  const [ongoingTxContractAddress, setOngoingTxContractAddress] = useState<string | null>(null)
   const [accountAddress, setAccountAddress] = useState<string | null>(null)
   const [accountId, setAccountId] = useState<AccountId | null>(null)
-  const { calculateHoldings, getLpTVL } = useFoxEthLiquidityPool(accountAddress)
-
-  const [farmingLoading, setFarmingLoading] = useState<boolean>(true)
-  const [foxFarmingTotalBalance, setFoxFarmingTotalBalance] = useState<string>('')
-  const [foxFarmingOpportunities, setFoxFarmingOpportunities] = useState<
-    FoxFarmingEarnOpportunityType[]
-  >([v4FarmingOpportunity, v3FarmingOpportunity, v2FarmingOpportunity, v1FarmingOpportunity])
-  const { lpApr } = useLpApr()
-  const ethMarketData = useAppSelector(state => selectMarketDataById(state, ethAssetId))
-  const foxMarketData = useAppSelector(state => selectMarketDataById(state, foxAssetId))
-  const foxAssetPrecision = useAppSelector(state => selectAssetById(state, foxAssetId)).precision
-  const ethAssetPrecision = useAppSelector(state => selectAssetById(state, ethAssetId)).precision
-  const lpAssetPrecision = useAppSelector(state =>
-    selectAssetById(state, foxEthLpAssetId),
-  ).precision
-  const [lpTokenPrice, setLpTokenPrice] = useState<string | null>(null)
-  const featureFlags = useAppSelector(selectFeatureFlags)
-
-  // TODO: Remove this useEffect
-  // The reason why it is still here is because we use accountAddress both for the modals, and to display the DeFi cards/rows
-  // If we remove it now, we will be unable to get to the modal in the first place to change accounts
-  useEffect(() => {
-    if (
-      // get price if at least one of lp or farming were on
-      (featureFlags.FoxLP || featureFlags.FoxFarming) &&
-      ethAssetPrecision &&
-      ethMarketData.price &&
-      lpAssetPrecision
-    ) {
-      ;(async () => {
-        try {
-          const lpTokenPrice = await getLpTokenPrice(
-            ethAssetPrecision,
-            ethMarketData.price,
-            lpAssetPrecision,
-          )
-          setLpTokenPrice(lpTokenPrice)
-        } catch (error) {
-          moduleLogger.error(error, 'getLpTokenPrice failed')
-        }
-      })()
-    }
-  }, [
-    ethAssetPrecision,
-    ethMarketData.price,
-    featureFlags.FoxFarming,
-    featureFlags.FoxLP,
-    lpAssetPrecision,
-  ])
+  const reduxDispatch = useDispatch()
```

Comment:
```suggestion
  const dispatch = useDispatch()
```

Also, using `useAppDispatch` instead will bring typings over and should be preferred